bin/programs/client/src/l2/executor/mod.rs

```diff
@@ -275,6 +275,49 @@ where
         Ok(&self.parent_header)
     }
 
+    /// Computes the current output root of the executor, based on the parent header and the
+    /// state's underlying trie.
+    ///
+    /// **CONSTRUCTION:**
+    /// ```text
+    /// output_root = keccak256(version_byte .. payload)
+    /// payload = state_root .. withdrawal_storage_root .. latest_block_hash
+    /// ```
+    ///
+    /// ## Returns
+    /// - `Ok(output_root)`: The computed output root.
+    /// - `Err(_)`: If an error occurred while computing the output root.
+    pub fn compute_output_root(&mut self) -> Result<B256> {
+        const OUTPUT_ROOT_VERSION: u8 = 0;
+        const L2_TO_L1_MESSAGE_PASSER_ADDRESS: Address =
```

Comment:
This should be grabbed from the superchain-registry no?

Reply:
Yeah good callout. Need some predeploy address constants in there, then can do! Will wait to merge, looks like there are some open PRs over there.